services/frontend-service/src/legacy-ui/App/index.tsx

```diff
@@ -52,16 +52,37 @@ export const Spinner: React.FC<any> = (props: any) => {
 };
 
 const GetOverview = (props: { children: (r: api.GetOverviewResponse) => JSX.Element }): JSX.Element | null => {
-    const getOverview = React.useCallback(
-        (api: any, authHeader: AuthHeaderType) => api.overviewService().StreamOverview({}, authHeader),
-        []
-    );
-    const overview = useObservable<api.GetOverviewResponse>(getOverview);
+    const [overview, setOverview] = useState<UnaryState<api.GetOverviewResponse>>({ state: 'pending' });
+    const Tdelay = 10 * 1000;
+    const api = React.useContext(Context);
+    const { authHeader } = React.useContext(AuthTokenContext);
+
+    const updateOverview = React.useCallback(() => {
+        api.overviewService()
+            .GetOverview({}, authHeader)
+            .then(
+                (result) => setOverview({ result, state: 'resolved' }),
+                (error) => setOverview({ error, state: 'rejected' })
+            );
+    }, [api, authHeader]);
+
+    React.useEffect(() => {
+        const id = setInterval(updateOverview, Tdelay);
+        return () => clearInterval(id);
+    }, [Tdelay, updateOverview]);
+
+    const backupState = useRef<api.GetOverviewResponse>();
+    if (backupState.current === undefined) updateOverview();
+
     switch (overview.state) {
         case 'resolved':
-            return props.children(overview.result);
+            backupState.current = overview.result;
+            return props.children(backupState.current);
         case 'rejected':
-            return <div>Error: {JSON.stringify(overview.error)}</div>;
+            // eslint-disable-next-line no-console
+            console.log('2 restarting streamoverview due to error: ', overview.error);
```

Comment:
I'm fine with the code as it is. I still think we need to make sure that the state is reloaded when a cart submit is successful.

Reply:
reload on submit is a more difficult than it seems. the overview currently is always a passed along local variable. but to have anything like that it would require us to use some sort of context or store or any other similar solution. and by then it seems too tricky. we can talk about this